enterprise/utils.py

```diff
@@ -1381,6 +1389,127 @@ def enroll_user(enterprise_customer, user, course_mode, *course_ids, **kwargs):
+            license_uuid=license_uuid,
+            enterprise_course_enrollment=enterprise_course_enrollment,
+        )
+    return enterprise_course_enrollment, created
+
+
+def enroll_licensed_users_in_courses(enterprise_customer, licensed_users_info, discount=100.00):
+    """
+    Takes a list of licensed learner data and enrolls each learner in the requested courses.
+
+    Args:
+        - enterprise_customer: The EnterpriseCustomer (object) which is sponsoring the enrollment
+        - licensed_users_info: (list) An array of dictionaries, each containing information necessary to create a
+            licensed enterprise enrollment for a specific learner in a specified course run.
+            Example:
+                licensed_users_info: [
+                    {
+                        'email': 'newuser@test.com',
+                        'course_run_key': 'course-v1:edX+DemoX+Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    }
+                ]
+        - discount: (int) the discount offered to the learner for their enrollment. Subscription based enrollments
+            default to 100
+
+    Expected Return Values:
+        Results: {
+            successes: [{ 'email': <email>, 'course_run_key': <key>, 'user': <user object> } ... ],
+            pending: [{ 'email': <email>, 'course_run_key': <key>, 'user': <user object> } ... ],
+            failures: [{ 'email': <email>, 'course_run_key': <key> } ... ]
+        }
+    """
+    results = {
+        'successes': [],
+        'pending': [],
+        'failures': [],
+    }
+    for licensed_user_info in licensed_users_info:
+        user = User.objects.filter(email=licensed_user_info['email']).first()
+        try:
+            if user:
+                succeeded = enroll_user(
+                    enterprise_customer,
+                    licensed_user_info['email'],
+                    licensed_user_info['course_mode'],
+                    licensed_user_info['course_run_key']
+                )
+                if succeeded:
+                    results['successes'].append({
+                        'user': user,
+                        'email': licensed_users_info['email'],
+                        'course_run_key': licensed_user_info['course_run_key']
+                    })
+
+                    enterprise_customer_user = get_enterprise_customer_user(user.id, enterprise_customer.uuid)
+                    get_create_ent_enrollment(
+                        licensed_user_info.get('course_run_key'),
+                        enterprise_customer_user,
+                        license_uuid=licensed_user_info.get('license_uuid')
+                    )
+                else:
+                    results['failures'].append({
+                        'email': licensed_users_info['email'], 'course_run_key': licensed_user_info['course_run_key']
```

Comment:
maybe store the `licensed_user_info.get('course_run_key')` and use it in these two places?